src/Common/Domain/ValueObject/TransactionId.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright (c) 2018 Adshares sp. z o.o.
+ *
+ * This file is part of AdServer
+ *
+ * AdServer is free software: you can redistribute and/or modify it
+ * under the terms of the GNU General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * AdServer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AdServer. If not, see <https://www.gnu.org/licenses/>
+ */
+declare(strict_types = 1);
+
+namespace Adshares\Common\Domain\ValueObject;
+
+use Adshares\Common\Id;
+use InvalidArgumentException;
+
+final class TransactionId implements Id
+{
+    /** @var string */
+    private $transactionIdStringRepresentation;
+
+    private function __construct(string $string)
+    {
+        $this->transactionIdStringRepresentation = $string;
+    }
+
+    public static function random(): TransactionId
+    {
+        $nodeId = str_pad(dechex(random_int(0, 2047)), 4, '0', STR_PAD_LEFT);
+        $tranId = str_pad(dechex(random_int(0, 2047)), 8, '0', STR_PAD_LEFT);
+        $mesgId = str_pad(dechex(random_int(0, 2047)), 4, '0', STR_PAD_LEFT);
+
+        return self::fromString("{$nodeId}:{$tranId}:{$mesgId}");
+    }
+
+    public static function fromString(string $string): TransactionId
+    {
+        if (!self::isValid($string)) {
+            throw new InvalidArgumentException("'$string' is NOT a VALID TransactionId representation.");
+        }
```

Comment:
I think that much safe place to check if an object is valid is object's constructor. Especially when someone adds a new method/factory different than `fromString` and forgot to check it.